internal/action/interaction.go

```diff
@@ -48,6 +49,41 @@ func (b *Builder) InteractNPCWithCheck(npc npc.ID, isCompletedFn func(d game.Dat
 	})
 }
 
+func (b *Builder) InteractChest(name object.Name, isCompletedFn func(game.Data) bool, additionalSteps ...step.Step) *Chain {
+	return NewChain(func(d game.Data) []Action {
+		o, _ := d.Objects.FindOne(name)
+		pos := o.Position
+		_, keyFound := d.Items.Find("Key", item.LocationInventory)
+
+		if o.InteractType == object.InteractTypeLocked && !keyFound {
+			return []Action{
+				b.MoveToCoords(pos, step.StopAtDistance(7)),
+				NewStepChain(func(d game.Data) []step.Step {
+					steps := []step.Step{step.InteractChest(o.Name, func(d game.Data) bool { return true }), step.SyncStep(func(d game.Data) error {
+						event.Send(event.InteractedTo(event.Text(b.Supervisor, ""), int(name), event.InteractionTypeObject))
+						return nil
+					})}
+
+					return append(steps, additionalSteps...)
+				}),
+			}
+		} else {
+
+			return []Action{
+				b.MoveToCoords(pos, step.StopAtDistance(7)),
+				NewStepChain(func(d game.Data) []step.Step {
+					steps := []step.Step{step.InteractChest(o.Name, isCompletedFn), step.SyncStep(func(d game.Data) error {
+						event.Send(event.InteractedTo(event.Text(b.Supervisor, ""), int(name), event.InteractionTypeObject))
+						return nil
+					})}
+
+					return append(steps, additionalSteps...)
+				}),
+			}
+		}
```

Comment:
What about this? So we only move/real attempt to open it if we have keys, otherwise we just return nil, doing nothing.
I think it's okay skip emitting the event, since it's a single interaction item, so companions will not be able to interact it.

```suggestion
		o, _ := d.Objects.FindOne(name)
		_, keyFound := d.Items.Find(item.Key, item.LocationInventory)

		if o.InteractType == object.InteractTypeLocked && keyFound {
			return []Action{
				b.MoveToCoords(o.Position, step.StopAtDistance(7)),
				NewStepChain(func(d game.Data) []step.Step {
					steps := []step.Step{step.InteractObject(o.Name, isCompletedFn), step.SyncStep(func(d game.Data) error {
						event.Send(event.InteractedTo(event.Text(b.Supervisor, ""), int(name), event.InteractionTypeObject))
						return nil
					})}

					return append(steps, additionalSteps...)
				}),
			}
		}

		return nil
```

Reply:
Thanks for the suggestion. I already tried to just return nil which backfired hard. For some reason, when you just return nil, it screws up the other chain/steps, leading to weird behavior. 

I tested specifically with the lower_kurast_chestrun.go script and that just led to error after error, skipping normal chests and other weird stuff. With this, it theoretically should just skip interacting with the chest no matter how many times its detected but I could be wrong ... 

I'll test my implementation again today (without keys in inventory) since I already have a compiled version of Koolo with the changes and see if it indeed gets stuck on locked chests (since I'll be running ancient tunnels & stony tomb)

btw @hectorgimenez I'll mail you later (on your publically shown email) with something I want to discuss with you to optimize the commits & contribution to the repo